iam/snippets/snippets_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package snippets
+
+import (
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func TestViewGrantableRoles(t *testing.T) {
+	projectID := os.Getenv("GOLANG_SAMPLES_PROJECT_ID")
+	resource := "//cloudresourcemanager.googleapis.com/projects/" + projectID
+	viewGrantableRoles(resource)
+}
+
+func TestQueryTestablePermissions(t *testing.T) {
+	projectID := os.Getenv("GOLANG_SAMPLES_PROJECT_ID")
+	resource := "//cloudresourcemanager.googleapis.com/projects/" + projectID
+	queryTestablePermissions(resource)
+}
+
+func TestRoles(t *testing.T) {
+	rand.Seed(time.Now().UnixNano())
+	roleName := "gotest" + strconv.Itoa(rand.Intn(100000))
```

Comment:
Does the role name need to be globally unique? Can this reuse the same role name? Trying to insert randomness has unfortunately lead to flaky tests in the past. So, we try to make it non-random. If we do need a random name, please use the `gofrs/uuid` package.

Reply:
It needs to be unique because roles persist 30 days in the system even after being deleted, so you can't re-use the same role name. Changed to use a GUID instead.